pkg/controller-gen/main.go

```diff
@@ -200,25 +209,44 @@ func generateDeepcopy(groups map[string]bool, customArgs *cgargs.CustomArgs) err
 		return nil
 	}
 
-	deepCopyCustomArgs := &dp.CustomArgs{}
-
-	args := args.Default().WithoutDefaultFlagParsing()
-	args.CustomArgs = deepCopyCustomArgs
-	args.OutputBase = customArgs.OutputBase
-	args.OutputFileBaseName = "zz_generated_deepcopy"
-	args.GoHeaderFilePath = customArgs.Options.Boilerplate
+	deepCopyArgs := dpargs.New()
+	deepCopyArgs.OutputFile = "zz_generated_deepcopy.go"
+	deepCopyArgs.GoHeaderFile = customArgs.Options.Boilerplate
 
+	inputDirs := []string{}
 	for gv, names := range customArgs.TypesByGroup {
 		if !groups[gv.Group] {
 			continue
 		}
-		args.InputDirs = append(args.InputDirs, names[0].Package)
-		deepCopyCustomArgs.BoundingDirs = append(deepCopyCustomArgs.BoundingDirs, names[0].Package)
+		inputDirs = append(inputDirs, names[0].Package)
+		deepCopyArgs.BoundingDirs = append(deepCopyArgs.BoundingDirs, names[0].Package)
+	}
+
+	getTargets := func(context *generator.Context) []generator.Target {
+		targets := dp.GetTargets(context, deepCopyArgs)
+
+		newTargets := []generator.Target{}
+		for _, t := range targets {
+			newTargets = append(newTargets, &generator.SimpleTarget{
+				PkgName:        t.Name(),
+				PkgPath:        t.Path(),
+				PkgDir:         t.Dir(),
+				HeaderComment:  t.Header(deepCopyArgs.OutputFile),
+				FilterFunc:     t.Filter,
+				GeneratorsFunc: t.Generators,
+			})
+		}
+
+		return newTargets
```

Comment:
You can just return the dp.GetTargets(...) from here if you want. I just did it because I had the need of change the dir at that point. But I'll approve it anyway.

Reply:
You're suggesting dropping the target modifications into newTargets and just returning dp.GetTargets directly?